roles/k3s/post/tasks/main.yml

```diff
@@ -16,6 +16,8 @@
     mode: 0755
   with_items: "{{ groups['master'] }}"
   run_once: true
+  notify: 
+    - restart-metallb-resources
```

Comment:
As `/tmp/k3s` is deleted in a later task, this will always return "changed". So even if we provision a completely fresh cluster, we will perform the (in this case unnecessary) restarts. So I propose using a better metric for determining whether we have changed the IP address pool, most likely the step where we apply the CRs.

Reply:
not sure, the template module should return changed=true only when the content of the template output is different then the current one on the remote, is it right?

Reply:
Correct. Still, [in a later task](https://github.com/techno-tim/k3s-ansible/blob/a1ef5904425785eec6a3bef467b5537c214943c6/roles/k3s/post/tasks/main.yml#L91), the directory containing this file is deleted. So the file will always be newly created (at least if the last play was not aborted) and thus the `template` module will always signal change. (This is of course not in accordance to best practices, but it is a different problem IMO).

Reply:
interesting. probably we should set some custom facts to deal with this? not sure